apps/kg/dataloader/sampler.py

```diff
@@ -8,6 +8,118 @@
 import pickle
 import time
 
+def SoftRelationPartition(edges, n, threshold=0.05):
+    """This partitions a list of edges based in to n partitions that 
+    SMALL relations (which has only small number of edges) will be put 
+    into a single LARGE partition (relations with large number of edges) 
+    will be evenly divided into all partitions.
```

Comment:
this statement is confusing.
This splits a list of edges into n partitions. Small relations will be placed into a single partition while a large relation will be evenly divided into all partitions.